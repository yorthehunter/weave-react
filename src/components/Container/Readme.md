Default Container
```js
<Container>
  <div style={{ outline: '1px solid red' }}>Content</div>
</Container>
```

Full Width Container
```js
<Container size="full-width">
  <div style={{ outline: '1px solid red' }}>Content</div>
</Container>
```

Container positioned right
```js
<Container position="right" size="xs">
  <div style={{ outline: '1px solid red' }}>Content</div>
</Container>
```

Container positioned left
```js
<Container position="left" size="xs">
  <div style={{ outline: '1px solid red' }}>Content</div>
</Container>
```

XS Container
```js
<Container size="xs">
  <div style={{ outline: '1px solid red' }}>Content</div>
</Container>
```

SM Container
```js
<Container size="sm">
  <div style={{ outline: '1px solid red' }}>Content</div>
</Container>
```

MD Container
```js
<Container size="md">
  <div style={{ outline: '1px solid red' }}>Content</div>
</Container>
```

LG Container
```js
<Container size="lg">
  <div style={{ outline: '1px solid red' }}>Content</div>
</Container>
```
