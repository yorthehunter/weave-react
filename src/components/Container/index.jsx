import React from 'react';
import { oneOf, string, node } from 'prop-types';
import 'weave-ui/src/components/container/index.scss';
import CustomTag from '../CustomTag';
import { sizeModifier, positionModifier } from '../utilities/util';

const componentPrefix = 'weave-container';


/**
 * The Container contains content, which automatically pads and breaks down the
 * layout based on preset breakpoints.
 */

const Container = (props) => {
  const {
    size,
    position,
    children,
  } = props;

  const componentClasses = [
    componentPrefix,
    sizeModifier(size, componentPrefix),
    positionModifier(position, componentPrefix),
  ];

  return (
    <CustomTag componentClasses={componentClasses} {...props} >
      { children }
    </CustomTag>
  );
};

const defaultProps = {
  tag: 'div',
  position: 'center',
  size: null,
  children: null,
};

const propTypes = {
  tag: string,
  position: oneOf([
    'left',
    'right',
    'center',
  ]),
  size: oneOf([
    'xs',
    'sm',
    'md',
    'lg',
    'full-width',
    null,
  ]),
  children: node,
};

Container.defaultProps = defaultProps;
Container.propTypes = propTypes;
export default Container;
