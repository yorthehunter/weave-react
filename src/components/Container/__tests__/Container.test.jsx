import React from 'react';
import renderer from 'react-test-renderer';

import Container from '../';

describe('It renders correctly', () => {
  it('with default values', () => {
    const rendered = renderer.create(<Container />);
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  it('with a custom span tag', () => {
    const props = {
      tag: 'span',
    };

    const rendered = renderer.create(<Container {...props} />);
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  it('with values set for xs size and left positioning', () => {
    const props = {
      size: 'xs',
      position: 'left',
    };

    const rendered = renderer.create(<Container {...props} />);
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
