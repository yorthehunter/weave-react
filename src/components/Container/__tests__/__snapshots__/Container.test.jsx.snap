// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders correctly with a custom span tag 1`] = `
<span
  className="weave-container"
  href={null}
  onClick={null}
  style={Object {}}
  type={null}
  value={null}
/>
`;

exports[`It renders correctly with default values 1`] = `
<div
  className="weave-container"
  href={null}
  onClick={null}
  style={Object {}}
  type={null}
  value={null}
/>
`;

exports[`It renders correctly with values set for xs size and left positioning 1`] = `
<div
  className="weave-container weave-container--xs weave-container--left"
  href={null}
  onClick={null}
  style={Object {}}
  type={null}
  value={null}
/>
`;
